tests/components/modbus/test_init.py

```diff
@@ -177,114 +190,202 @@ async def test_config_multiple_modbus(hass, caplog):
             CONF_PORT: "usb01",
             CONF_PARITY: "E",
             CONF_STOPBITS: 1,
-            CONF_NAME: "modbusTest3",
+            CONF_NAME: TEST_MODBUS_NAME + "3",
         },
     ]
 
-    caplog.set_level(logging.ERROR)
-    await _config_helper(hass, do_config)
-    assert DOMAIN in hass.config.components
-    assert len(caplog.records) == 0
+    await _config_helper(hass, do_config, caplog)
 
 
-async def test_pb_service_write_register(hass):
+async def test_pb_service_write_register(hass, caplog, mock_modbus):
     """Run test for service write_register."""
 
-    conf_name = "myModbus"
+    # Pymodbus write single, response OK.
+    data = {ATTR_HUB: TEST_MODBUS_NAME, ATTR_UNIT: 17, ATTR_ADDRESS: 16, ATTR_VALUE: 15}
+    await hass.services.async_call(DOMAIN, SERVICE_WRITE_REGISTER, data, blocking=True)
+    assert mock_modbus.write_register.called
+    assert mock_modbus.write_register.call_args[0] == (
+        data[ATTR_ADDRESS],
+        data[ATTR_VALUE],
+    )
+
+    # Pymodbus write single, response error or exception
+    caplog.set_level(logging.DEBUG)
+    mock_modbus.write_register.return_value = ExceptionResponse(0x06)
+    await hass.services.async_call(DOMAIN, SERVICE_WRITE_REGISTER, data, blocking=True)
+    assert caplog.messages[-1].startswith("Pymodbus:")
```

Comment:
I'd assert the calls to the mock also.

Reply:
you are right, it is a little more secure. I hope I added it everywhere this time :-)

Reply:
We can reset the mock in between also, so that the calls are reset.

https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.reset_mock

Reply:
I knew there was something missing, added.